### PR TITLE
Improve Contact.fromString parsing and Account.meUsingAlias fallback

### PR DESCRIPTION
**Pull request by bengotow**
## Summary

Improves the robustness of contact string parsing to handle malformed user input (particularly account aliases) without throwing errors. Updates `Contact.fromString()` to gracefully handle multiple email addresses and missing emails, and adds fallback logic to `Account.meUsingAlias()` to ensure a valid email address is always present.

## Key Changes

- **Contact.fromString() parsing improvements**:
  - Changed from throwing an error when multiple emails are found to intelligently selecting the best match
  - Prefers email addresses wrapped in angle brackets (e.g., `Name <email@example.com>`)
  - Falls back to the last email found when no bracketed address exists
  - Gracefully handles strings with no email address by returning empty email and using the entire string as the name
  - Added comprehensive JSDoc explaining the parsing strategy and why it must never throw

- **Account.meUsingAlias() fallback logic**:
  - Added validation to check if parsed alias contains an email address
  - Falls back to the account's own email address if the alias has no email
  - Preserves the alias as a display name while using the account's email for sending
  - Ensures the contact always has a valid email address for mail operations

- **Test coverage**:
  - Added 6 new test cases for `Contact.fromString()` covering edge cases: well-formed input, bare emails, multiple addresses, missing emails, and empty strings
  - Added 5 new test cases for `Account.meUsingAlias()` covering empty aliases, well-formed aliases, duplicate emails, missing emails, and whitespace-only input

## Implementation Details

The parsing logic now collects all email matches in the string and applies selection heuristics:
1. Filter for emails wrapped in angle brackets
2. Use the last bracketed match if available
3. Otherwise use the last match found (since display name typically precedes email)
4. Extract the display name as everything before the chosen email, trimmed and cleaned of trailing delimiters

This approach handles common malformed alias patterns like `ben@mailspring.com <ben@mailspring.com>` (email as display name) without throwing, which is critical since a single bad alias would otherwise crash views that call `Contact#isMe()`.

https://claude.ai/code/session_013T1jsQixghcikuw6s3PiUy

### CODE REVIEW

File: app/src/flux/models/account.ts [hunk 1]
```diff
@@ -164,9 +164,20 @@ export class Account extends ModelWithMetadata {
...
     });
+
+    // Aliases are user-provided and may not contain an email address at all
+    // (eg: legacy or hand-edited data.) Treat the alias as a display name over
+    // the account's own address - which is what the preferences UI saves for
+    // the same input - rather than returning a contact that can't send mail.
+    if (!contact.email) {
+      contact.email = this.emailAddress;
+      contact.name = contact.name || this.name;
+      contact.id = `local-${this.id}-${this.emailAddress}`;
```
[the remark this answers is not in the document]
Comment (bengotow) [reply app/src/flux/models/account.ts:178]: Keeping the email-based id here — the suggested change would introduce the duplicate it's warning about.

There is already an id-keyed consumer in tree: the composer's From dropdown renders `AccountStore.aliasesFor(accounts)` through `Menu` with `itemKey={contact => contact.id}` (`app/internal_packages/composer/lib/account-contact-field.tsx:123`). Since `aliases()` pushes `acc.me()` and then every `meUsingAlias(alias)` result into the same list, reusing `local-${this.id}-me` for the fallback would put two items with an identical key in that menu — a React duplicate-key collision on the exact path this PR is trying to keep working. Avoiding that is why the id is rewritten rather than inherited from `me()`.

The email-based shape is also the one this alias would already have: `local-${accountId}-${email}` is what `Contact.fromString` produces for the well-formed equivalent (`Bob <account@email.com>`), which is what the preferences UI saves for the input `Bob`. So the fallback lands on the same id as the alias it stands in for, rather than a third shape. Two aliases sharing an email already collide on id today — see the "the user may have two aliases with the same email but different phrases" case in `draft-factory.ts:396` — so this isn't new ground.

Took the second finding: the `name.trim()` in the `Contact.fromString` constructor call was indeed a no-op, removed in 6815654.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/src/flux/models/contact.ts [hunk 2]
```diff
@@ -412,24 +412,43 @@ export class Contact extends Model {
...
+  // Note: This is called with user-provided values - account aliases in particular -
+  // which are often malformed. `ben@foundry376.com <ben@foundry376.com>` (the email
+  // used as the display name) and strings with no email address at all are both
+  // common. Parsing must never throw: a single bad alias is read by
+  // AccountStore.aliases(), and would otherwise take down every view that calls
+  // Contact#isMe, including the thread list.
+  //
+  // If no email address is present, the returned contact has an empty `email` and
+  // the entire string as its `name`.
+  static fromString(string: string, { accountId }: { accountId?: string } = {}) {
```
Comment (bengotow) [inline app/src/flux/models/contact.ts:426]: This looks good, I'm surprised there are no other callers to this outside this one account use case, but that seems to be the case.
